docs/proposals/20191017-kubeadm-based-control-plane.md

```diff
@@ -346,58 +387,73 @@ spec:
 - Completely removing the control plane and issuing a delete on the underlying machines.
 - User documentation should focus on deletion of the Cluster resource rather than the KubeadmControlPlane resource.
 
-##### KubeadmControlPlane rollout (using create-swap-and-delete)
+##### KubeadmControlPlane rollout
 
-- Triggered by: 
-    - Changes to Version
-    - Changes to the kubeadmConfigSpec
-    - Changes to the infrastructureRef
-    - The `upgradeAfter` field, which can be set to a specific time in the future
-        - Set to `nil` or the zero value of `time.Time` if no upgrades are desired
-        - An upgrade will run when that timestamp is passed
-        - Good for scheduling upgrades/SLOs
-        - Set `upgradeAfter` to now (in RFC3339 form) if an upgrade is required immediately
-
-- Rollout operations rely on scale up and scale down which are be blocked based on Etcd and control plane health checks
+KubeadmControlPlane rollout operations rely on [scale up](#scale up) and [scale down](#scale_down) which are be blocked based on Etcd and control plane health checks
   - See [Health checks](#Health checks) below.
 
+KubeadmControlPlane rollout is triggered by:
+  - Changes to Version
+  - Changes to the kubeadmConfigSpec
+  - Changes to the infrastructureRef
+  - The `upgradeAfter` field, which can be set to a specific time in the future
+    - Set to `nil` or the zero value of `time.Time` if no upgrades are desired
+    - An upgrade will run when that timestamp is passed
+    - Good for scheduling upgrades/SLOs
+    - Set `upgradeAfter` to now (in RFC3339 form) if an upgrade is required immediately
+
 - The rollout algorithm is the following:
   - Find Machines that have an outdated spec
   - If there is a machine requiring rollout
     - Scale up control plane creating a machine with the new spec
     - Scale down control plane by removing one of the machine that needs rollout (the oldest out-of date machine in the failure domain that has the most control-plane machines on it)
-    
+
+- The controller should tolerate the manual or automatic removal of a replica during the upgrade process. A replica that fails during the upgrade may block the completion of the upgrade. Removal or other remedial action may be necessary to allow the upgrade to complete.
+
+##### KubeadmControlPlane rollout (using create-swap-and-delete)
```

Comment:
```suggestion
##### RolloutType ScaleOut (create-swap-and-delete)
```
Also, @jan-est what about:
- moving L405-L409 down into RolloutType ScaleOut (it is the description of scale out)
- moving L417-L421 up into the main Rollout chapter (this is how we determine a machine need rollout, which applies to both strategies)

Reply:
Done